Makefile

```diff
@@ -0,0 +1,48 @@
+.PHONY: help
+
+#
+# Configuration variables
+#
+VENVDIR?=.venv
+VENV = $(VENVDIR)/bin
+
+PYTHON3OK :=$(shell python3 --version 2>&1)
+ifeq ($(PYTHON3OK),)
+  brew -y install python3.6 python3-pip
+endif
+
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+.PHONY: venv
+venv: ## create virtual environment
+		test -d $(VENV) || python3 -m venv $(VENVDIR)
+
+.PHONY: activate
+activate: venv	## activate venv
```

Comment:
the `activate` target as well as `show-venv`, `venv-activate` and `clean-venv` are not necessary.

we should only have purpose oriented targets that achieve an actual outcome, like `test` or `report`. The `venv` is just a sand-boxing utility to make sure we have a repeatable setup and do not mess with the user environment.